hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngradeUtils.java

```diff
@@ -203,7 +205,8 @@ static void rollbackFailedWritesAndCompact(HoodieTable table, HoodieEngineContex
         if (shouldCompact) {
           Option<String> compactionInstantOpt = writeClient.scheduleCompaction(Option.empty());
           if (compactionInstantOpt.isPresent()) {
-            writeClient.compact(compactionInstantOpt.get());
+            HoodieWriteMetadata result = writeClient.compact(compactionInstantOpt.get());
+            writeClient.commitCompaction(compactionInstantOpt.get(), result, Option.empty());
```

Comment:
add inline completion methods and use that?

Reply:
responded above.